onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -3883,62 +3883,37 @@ def aten_matrix_power(self: TensorType, n: int) -> TensorType:
     raise NotImplementedError()
 
 
-@torch_op("aten::max", trace_only=True)
-def aten_max(
-    self: TReal,
-    dim_or_other: Optional[Union[TReal, INT64]] = None,
-    keepdim: Optional[BOOL] = None,
-) -> TReal:
+@torch_op("aten::max")
+def aten_max(self: TReal) -> TReal:
     """max(Tensor self) -> Tensor"""
 
     self_rank = op.Size(op.Shape(self))
     if self_rank == 0:
-        self = op.Reshape(self, op.Constant(value_int=[-1]))
-
-    output = 1
-
-    if op.OptionalHasElement(dim_or_other):
-        if isinstance(dim_or_other, int):
-            if not op.OptionalHasElement(keepdim):
-                keepdim = False
-            result, indices = _aten_max_with_dim(self, dim_or_other, keepdim)
-            output = 2
-        else:  # dim_or_other is tensor
-            result = _aten_max_with_other(self, dim_or_other)
-    else:
-        result = _aten_max_with_no_dim(self)
+        self = op.Reshape(self, op.Constant(value_ints=[-1]))
+
+    result = op.ReduceMax(self, keepdims=0)
 
     if self_rank == 0:
         result = op.Squeeze(result)
 
-    if output == 2:
-        if self_rank == 0:
-            indices = op.Squeeze(indices)  # type: ignore[has-type]
-        return result, indices
-    return result
-
-
-@torch_op("aten::max", private=True)
-def _aten_max_with_no_dim(self: TReal) -> TReal:
-    result = op.ReduceMax(self, keepdims=0)
     return result
 
 
-@torch_op("aten::max", private=True)
-def _aten_max_with_other(self: TReal, other: TReal) -> TReal:
-    result = op.Max(self, other)
-    return result
-
+@torch_op("aten::max.dim")
+def aten_max_dim(self: TReal, dim: int, keepdim: bool = False) -> Tuple[TReal, INT64]:
```

Comment:
Downstairs on line 3932, there is a mean.dim, I think these two have a pretty similar implementation, but that one seems to be wrongly annotated with dim: INT64. Wonder why is that one passed?

Reply:
Thanks for catching! I updated the function. Let me check the test

Reply:
OK so it was actually correct. That is because dim in `mean_dim` is `int[1]? dim`, which is a sequence that is passed as an input. We treat it as INT64 to feed it into the ReduceMean op.

Reply:
Did test fail? I thought every parameters went to attribute of an Op, should be primitive type. If it's list of ints, shouldn't it be Sequence[int]?

Reply:
Ah I see there is a 

```python
 op.Size(op.Shape(dim)) == 0:
```

Reply:
I am not sure if there's an easy way to process `Sequence[int]` from an attribute to make it an input. ConcatFromSequence apparently doesn't work plus it is going to be slower. So we make it an input right from the beginning instead.